dask/dataframe/io/parquet/fastparquet.py

```diff
@@ -1187,16 +1196,22 @@ def write_partition(
             return []
 
     @classmethod
-    def write_metadata(cls, parts, fmd, fs, path, append=False, **kwargs):
+    def write_metadata(
+        cls, parts, fmd, fs, path, append=False, metadata=None, **kwargs
+    ):
         _meta = copy.copy(fmd)
+        rgs = fmd.row_groups
         if parts:
             for rg in parts:
                 if rg is not None:
                     if isinstance(rg, list):
                         for r in rg:
-                            _meta.row_groups.append(r)
+                            rgs.append(r)
                     else:
-                        _meta.row_groups.append(rg)
+                        rgs.append(rg)
+            _meta.row_groups = rgs
+            if metadata:
+                _meta
```

Comment:
Can you explain this?

Reply:
Now this one is definitely a mistake. I assume it should be a copy of the lines to extend the metadata, but I wonder why no test caught it. I'd have to look into what `metadata` is supposed to contain at this point.

Reply:
This line is never touched in the code, seems we never pass `metadata`, I'll remove this.